pycalphad/core/phase_rec.pxd

```diff
@@ -48,20 +80,11 @@ cdef public class PhaseRecord(object)[type PhaseRecordType, object PhaseRecordOb
     cpdef void internal_cons_func(self, double[::1] out, double[::1] dof) nogil
     cpdef void internal_cons_jac(self, double[:,::1] out, double[::1] dof) nogil
     cpdef void internal_cons_hess(self, double[:,:,::1] out, double[::1] dof) nogil
+    cpdef void phase_local_cons_func(self, double[::1] out, double[::1] dof, FastFunction func) nogil
+    cpdef void phase_local_cons_jac(self, double[:, ::1] out, double[::1] dof, FastFunction jac_func) nogil
     cpdef void mass_obj(self, double[::1] out, double[::1] dof, int comp_idx) nogil
     cpdef void mass_obj_2d(self, double[::1] out, double[:, ::1] dof, int comp_idx) nogil
     cpdef void formulamole_obj(self, double[::1] out, double[::1] dof, int comp_idx) nogil
     cpdef void formulamole_grad(self, double[::1] out, double[::1] dof, int comp_idx) nogil
     cpdef void formulamole_hess(self, double[:,::1] out, double[::1] dof, int comp_idx) nogil
-    # Used only to reconstitute if pickled (i.e. via __reduce__)
-    cdef public object ofunc_
-    cdef public object formulaofunc_
-    cdef public object formulagfunc_
-    cdef public object formulahfunc_
-    cdef public object internal_cons_func_
-    cdef public object internal_cons_jac_
-    cdef public object internal_cons_hess_
-    cdef public object massfuncs_
-    cdef public object formulamolefuncs_
-    cdef public object formulamolegradfuncs_
-    cdef public object formulamolehessianfuncs_
+
```

Comment:
We talked offline and the cache in `FastFunctionFactory` is not persisted through pickle. This could potentially be okay, but we should investigate further.